packages/web3-utils/src/converters.ts

```diff
@@ -436,3 +437,56 @@ export const jsonInterfaceMethodToString = (
 
 	return `${json.name}(${flattenTypes(false, json.inputs).join(',')})`;
 };
+
+export const convertToValidType = (
+	value: string | number | bigint,
+	desiredType: ValidTypes,
+): ValidReturnTypes[ValidTypes] => {
+	switch (desiredType) {
+		case ValidTypes.HexString:
+			return toHex(value);
+		case ValidTypes.NumberString:
+			return hexToNumberString(toHex(value));
+		case ValidTypes.Number:
+			return toNumber(value);
+		case ValidTypes.BigInt:
+			return BigInt(toNumber(value));
+		default:
+			// TODO Replace with proper error
+			throw new Error('Invalid desiredType provided');
```

Comment:
TODO

Reply:
This code needs to have extensive tests coverage with variety of input values.

Reply:
These tests have been written [here](https://github.com/ChainSafe/web3.js/pull/4525/files#diff-f8ab0cde9e2a4da690869725ec95f381288fd4ff11dc3b0df8081e73d1e9d36bR274)